autogpt/main.py

```diff
@@ -81,17 +81,16 @@ def run_auto_gpt(
     if install_plugin_deps:
         install_plugin_dependencies()
 
-    # TODO: have this directory live outside the repository (e.g. in a user's
-    #   home directory) and have it come in as a command line argument or part of
-    #   the env file.
-    if workspace_directory is None:
-        workspace_directory = Path(__file__).parent / "auto_gpt_workspace"
-    else:
+    if workspace_directory:
```

Comment:
First priority is command line argument
Second is environment variable
The last is default

Reply:
I need help finding the place where the CFG.workspace_path gets updated if the command line argument is passed in.

By line 104 below, it seems possible that CFG.workspace_path is out of sync with "workspace_directory" and because it's still pointing to what was set up on line 20 of config.py above.

It's probably better to have the config go in one direction - either autogpt.main (and Workspace) reads from Config, or vice versa.

Elsewhere in autogpt.main, I see us calling create_config(...) maybe we inject the workspace path there and then have Workspace read it's config from the config object.